dcos/subcommand.py

```diff
@@ -219,8 +226,74 @@ def _write_package_json(pkg):
         json.dump(package_json, package_file)
 
 
-def _install_env(pkg, options):
-    """ Install subcommand virtual env.
+def _hashfile(filename):
+    """Calculates the sha256 of a file
+
+    :param filename: path to the file to sum
+    :type filename: str
+    :returns: digest in hexadecimal
+    :rtype: str
+    """
+
+    hasher = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        for chunk in iter(lambda: f.read(4096), b''):
+            hasher.update(chunk)
+    return hasher.hexdigest()
+
+
+def _check_hash(filename, content_hashes):
+    """Validates whether downloaded binary matches expected hash
+
+    :param filename: path to binary
+    :type filename: str
+    :param content_hashes: list of hash algorithms/value
+    :type content_hashes: [{}]
+    :returns: None if valid hash, else throws exception
+    :rtype: None
+    """
+
+    content_hash = next((contents for contents in content_hashes
+                        if contents.get("algo") == "sha256"),
+                        None)
+    if content_hash:
+        expected_value = content_hash.get("value")
+        if expected_value != _hashfile(filename):
+            raise DCOSException(
+                "Downloaded CLI contents do not match expected contents. "
+                "Aborting...")
+        else:
+            return
+    else:
+        raise DCOSException(
+            "Hash algorithm specified is unsupported. "
+            "Please contact the project maintainer")
+
+
+def _get_cli_binary(resources):
```

Comment:
it returns `{"url": <str>, "kind": <str>, "contentHash": [{}]}`, so updated pydoc and renamed to `_get_cli_binary_info`